dll/win32/iernonce/registry.cpp

```diff
@@ -0,0 +1,206 @@
+/*
+ * PROJECT:     ReactOS system libraries
+ * LICENSE:     GPL - See COPYING in the top level directory
+ * FILE:        dll\win32\iernonce\registry.cpp
+ * PURPOSE:     Functions to read RunOnceEx registry.
+ * PROGRAMMERS: Copyright 2013-2016 Robert Naumann
+ */
+
+
+#include <cassert>
+#include "registry.h"
+#include <winreg.h>
+
+
```

Comment:
This looks like quite over-engineering for just enumerating subkeys and values.
Also if you use C++ stuff with ATL, you could directly use the CRegistryKey class.